src/kubectl-object-model.ts

```diff
@@ -122,6 +133,28 @@ export class Kubectl {
             return this.execute(['delete'].concat(args));
     }
 
+    public getDeployedObjects() {
+        let manifests = getManifestObjects(this.deployedManifestsPaths);
+        let deployedObjects: IKubeObject[] = [];
+        if (manifests &&
+            manifests.length > 0) {
+                manifests.forEach((manifestContent) => {
+                    if (manifestContent &&
+                        manifestContent.apiVersion &&
+                        manifestContent.kind &&
+                        manifestContent.metadata &&
+                        manifestContent.metadata.name) {
+                            deployedObjects.push({
+                                version: manifestContent.apiVersion,
+                                kind: manifestContent.kind,
+                                name: manifestContent.metadata.name
+                            });
+                        }
+                });
+            }
+        return deployedObjects;

```

Comment:
We should ensure uniqueness of these objects.
In some cases like `canary promote`, `Trafficsplit` is configured twice in the action lifetime. There may be other cases.